lib/api/docs_search_options.rb

```diff
@@ -6,6 +6,7 @@ class Api::DocsSearchOptions < Api::CommercialSearchOptions
                         message: 'must be present'
 
   def initialize(params = {})
+    params = params.to_h unless params.is_a?(Hash)
```

Comment:
This is another case where it would be better to convert the params to a hash further up the code path. In this case, the params being passed to this initialization are assembled in Api::V2::SearchesController#search_params. Converting the params to a hash in that method will resolve the deprecation warning in this file, as well as any other potential downstream issues:
```ruby
#app/controllers/api/v2/searches_controller.rb

      def search_params
        @search_params ||= params.permit(:access_key,
                                         :affiliate,
                                         :dc,
                                         :api_key,
                                         :cx,
                                         :enable_highlighting,
                                         :format,
                                         :limit,
                                         :offset,
                                         :query,
                                         :sort_by,
                                         :sc_access_key,
                                         :routed,
                                         :query_not,
                                         :query_quote,
                                         :query_or,
                                         :filetype,
                                         :filter
                                        ).to_h

      end
```